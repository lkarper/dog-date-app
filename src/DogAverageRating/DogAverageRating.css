.DogAverageRating__star-outer-container {
    display: flex;
    flex-direction: column;
}
@media screen and (min-width: 750px) {
    .DogAverageRating__star-outer-container {
        flex-direction: row;
        align-items: center;
    }
}
