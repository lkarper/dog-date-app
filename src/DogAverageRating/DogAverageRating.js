import React from 'react';

export const calculateAverageWithArrayOfReviews = (reviews) => {
    return reviews.reduce((acc, curr) => {
        const { 
            friendliness_dogs,
            friendliness_people,
            playing_interest,
            obedience,
            profile_accuracy,
            location_suitability,
        } = curr;
        const averageRating = (
            friendliness_dogs +
            friendliness_people +
            playing_interest +
            obedience +
            profile_accuracy + 
            location_suitability 
        ) / 6;
        return acc + averageRating;
    }, 0) / reviews.length;
}

const DogAverageRating = (props) => {
    const { reviews } = props;

    const totalAverage = calculateAverageWithArrayOfReviews(reviews);

    return (
        <p>Average rating: {totalAverage.toFixed(2)} stars</p>
    );

}

export default DogAverageRating;