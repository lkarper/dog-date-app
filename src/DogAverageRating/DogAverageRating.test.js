import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import DogAverageRating from './DogAverageRating';

describe('DogAverageRating component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<DogAverageRating />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<DogAverageRating />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
