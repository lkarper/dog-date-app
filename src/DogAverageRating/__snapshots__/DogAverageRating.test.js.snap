// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DogAverageRating component renders the UI as expected 1`] = `
<DogAverageRating
  reviews={
    Array [
      Object {},
    ]
  }
>
  <p>
    Average rating not available.
  </p>
</DogAverageRating>
`;
